hack/e2e-kind-cluster-setup.sh

```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+set -xe
+
+CNI_VERSION=${CNI_VERSION:-0.4.0}
+OCI_BIN=${OCI_BIN:-docker}
+IMG_REGISTRY=${IMAGE_REGISTRY:-localhost:5000/maiqueb}
+
+setup_cluster() {
+    git clone https://github.com/k8snetworkplumbingwg/multus-cni/
+    pushd multus-cni/e2e
+    trap "popd" RETURN SIGINT
+    ./get_tools.sh
+    CNI_VERSION="$CNI_VERSION" ./generate_yamls.sh
+    OCI_BIN="$OCI_BIN" ./setup_cluster.sh
+}
+
+push_local_image() {
+    OCI_BIN="$OCI_BIN" IMAGE_REGISTRY="$IMG_REGISTRY" make manifests
+    OCI_BIN="$OCI_BIN" IMAGE_REGISTRY="$IMG_REGISTRY" make img-build
+    "$OCI_BIN" push "$IMG_REGISTRY/multus-dynamic-networks-controller:latest"
+}
+
+cleanup() {
+    rm -rf multus-cni
+    git checkout -- manifests/
+}
+
+trap "cleanup" EXIT
```

Comment:
you might want to add here SIGINT as well (unless you remove the other one)

Reply:
Why ? It still works:

```
sudo --preserve-env=PATH OCI_BIN=podman hack/e2e-kind-cluster-setup.sh
+ CNI_VERSION=0.4.0
+ OCI_BIN=podman
+ IMG_REGISTRY=localhost:5000/maiqueb
+ trap cleanup EXIT
+ setup_cluster
+ git clone https://github.com/k8snetworkplumbingwg/multus-cni/
Cloning into 'multus-cni'...
remote: Enumerating objects: 38696, done.
remote: Counting objects: 100% (1263/1263), done.
remote: Compressing objects: 100% (749/749), done.
remote: Total 38696 (delta 458), reused 1208 (delta 439), pack-reused 37433
Receiving objects: 100% (38696/38696), 49.67 MiB | 12.64 MiB/s, done.
Resolving deltas: 100% (17230/17230), done.
+ pushd multus-cni/e2e
/home/mduarted/github/multus-dynamic-networks-controller/multus-cni/e2e /home/mduarted/github/multus-dynamic-networks-controller
+ trap popd RETURN SIGINT
+ ./get_tools.sh
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 6511k  100 6511k    0     0  7275k      0 --:--:-- --:--:-- --:--:-- 32.7M
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 42.9M  100 42.9M    0     0  37.6M      0  0:00:01  0:00:01 --:--:-- 37.6M
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0^C++ popd
/home/mduarted/github/multus-dynamic-networks-controller
+ cleanup
+ rm -rf multus-cni
+ git checkout -- manifests/
```

Reply:
yes because it is the SIGINT in the function that catched here
assuming that this trap live only in the function, what if SIGINT happening out side of the function on machines where SIGINT is needed ?

anyway its not that important, but just for the discussion